diff --git a/src/router.js b/src/router.js
index 5b76ccfc046a5c8ee49a9b88e03d8d4476b5e4a9..1bb8a992d365cf629100b4fceef02c9e2481e8d2 100644
--- a/src/router.js
+++ b/src/router.js
@@ -185,7 +185,7 @@ export function Router(props) {
 
 		// Only mark as an update if the route component changed.
 		const outgoing = prev.current && prev.current.props.children;
-		if (!outgoing || !incoming || incoming.type !== outgoing.type || incoming.props.component !== outgoing.props.component) {
+		if (!outgoing || !incoming || incoming.type !== outgoing.type || incoming.props.component !== outgoing.props.component || incoming.props.remount) {
 			// This hack prevents Preact from diffing when we swap `cur` to `prev`:
 			if (this.__v && this.__v.__k) this.__v.__k.reverse();
 			count.current++;
